backend/routing/daemon_controller.py

```diff
@@ -149,7 +155,53 @@ async def force_downgrade(
             logger=request.logger,
             regenerate_from=regenerate_from,
         )
-                
+
+    async def bulk_process_file_background(
+        self,
+        files_repo: FileRepository,
+        passthrough_request: Request,
+        files: List[FileModel],
+        stop_at: DocumentStatus,
+        regenerate_from: DocumentStatus,
+        max_documents: Optional[int] = None,
+    ) -> None:
+        logger = passthrough_request.logger
+        if max_documents is None:
+            max_documents = -1
+        if files is None:
+            logger.info("List of files to process was empty")
+            return None
+        if len(files) == 0:
+            logger.info("List of files to process was empty")
+            return None
+        for file in files:
+            if max_documents == 0:
+                logger.info("Reached maxiumum file limit, exiting.")
+                return None
+            logger.info(f"Validating file {str(file.id)} for processing.")
+            file_stage = DocumentStatus(file.stage)
+            if docstatus_index(file_stage) > docstatus_index(regenerate_from):
+                file_stage = regenerate_from
+                file.stage = regenerate_from.value
+                await files_repo.update(file)
+                await files_repo.session.commit()
+                logger.info(
+                    f"Reverting fileid {
+                        file.id} to stage {file.stage}"
+                )
+            # Dont process the file if it is already processed beyond the stop point.
+            if docstatus_index(file_stage) < docstatus_index(stop_at):
+                logger.info(
+                    f"Sending file {
+                        str(file.id)} to be processed in the background."
+                )
+                # copy_files_repo = copy.deepcopy(files_repo)
+                max_documents += -1
+                passthrough_request.app.emit(
+                    "process_document",
+                    doc_id_str=str(file.id),
+                    stop_at=stop_at.value,
+                )
```

Comment:
This func was deleted on purpose, since adding documents is done by popping the fileid onto the processing queue, it shouldnt be getting called by anything